storage/watchable_store.go

```diff
@@ -320,12 +330,18 @@ func (s *watchableStore) notify(rev int64, evs []storagepb.Event) {
 		if eb.revs != 1 {
 			panic("unexpected multiple revisions in notification")
 		}
+		w.mu.Lock()
+		id := w.id
+		cur := w.cur
+		w.mu.Unlock()
 		select {
-		case w.ch <- WatchResponse{WatchID: w.id, Events: eb.evs, Revision: s.Rev()}:
+		case w.ch <- WatchResponse{WatchID: id, Events: eb.evs, Revision: s.Rev(), State: WatchState{StartRevision: cur}}:
 			pendingEventsGauge.Add(float64(len(eb.evs)))
 		default:
 			// move slow watcher to unsynced
+			w.mu.Lock()
 			w.cur = rev
```

Comment:
You are right. So we don't update `cur` in synced. Only when it's unsynced, we change `w.cur`  as far as I understand.